signature/decode_hex.go

```diff
@@ -0,0 +1,13 @@
+package signature
+
+import (
+	"encoding/hex"
+	"strings"
+)
+
+func DecodeHex(str string) ([]byte, error) {
+	if strings.HasPrefix(str, "0x") {
+		return hex.DecodeString(strings.TrimPrefix(str, "0x"))
+	}
+	return hex.DecodeString(str)
```

Comment:
```suggestion
	return hex.DecodeString(strings.TrimPrefix(str, "0x"))
```
you actually don't need to check for `0x`, it will be handled by strings library:

```
// TrimPrefix returns s without the provided leading prefix string.
// If s doesn't start with prefix, s is returned unchanged.
```



```